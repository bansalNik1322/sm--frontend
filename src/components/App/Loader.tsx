'use client';

import React, { useEffect, useState } from 'react';
import Spinner from 'react-bootstrap/Spinner';
import { Bars, ThreeDots } from 'react-loader-spinner';

function ButtonLoader() {
  return (
    <div className="d-flex justify-content-center">
      <ThreeDots
        height="30"
        width="30"
        color="#FFFFFF"
        ariaLabel="three-circles-rotating"
        wrapperStyle={{}}
        wrapperClass=""
        visible={true}
      />
    </div>
  );
}

export function BarLoader() {
  return (
    <div className="d-flex justify-content-center">
      <Bars
        height="30"
        width="30"
        color="white"
        ariaLabel="three-circles-rotating"
        wrapperStyle={{}}
        wrapperClass=""
        visible={true}
      />
    </div>
  );
}

export function SuspenseLoader() {
  return (
    <div className="d-flex justify-content-center" style={{ marginTop: '5%' }}>
      <ThreeDots
        height="30"
        width="50"
        // color="var(--sp-primary)"
        color="white"
        ariaLabel="three-circles-rotating"
        wrapperStyle={{}}
        wrapperClass=""
        visible={true}
      />
    </div>
  );
}

export function SimpleLoader() {
  return <Spinner animation="border" style={{ color: 'white' }} />;
}
export const useLoading = () => {
  return { ButtonLoader, SimpleLoader };
};

export function Loading() {
  // const router = useRouter();
  // const asPath = usePathname();
  const [loading] = useState(false);
  useEffect(() => {
    // const handleStart = (url: string) => url !== asPath && setLoading(true);
    // const handleComplete = (url: string) => url === asPath && setLoading(false);
    // router.events.on('routeChangeStart', handleStart);
    // router.events.on('routeChangeComplete', handleComplete);
    // router.events.on('routeChangeError', handleComplete);
    // return () => {
    //   router.events.off('routeChangeStart', handleStart);
    //   router.events.off('routeChangeComplete', handleComplete);
    //   router.events.off('routeChangeError', handleComplete);
    // };
  });
  return loading ? (
    <div className="mainLoader">
      <div className="d-flex justify-content-center" style={{ marginTop: '25%' }}>
        <ThreeDots
          height="30"
          width="50"
          // color={'#002e6e'}
          color={'white'}
          ariaLabel="three-circles-rotating"
          wrapperStyle={{}}
          wrapperClass=""
          visible={true}
        />
      </div>
    </div>
  ) : null;
}
