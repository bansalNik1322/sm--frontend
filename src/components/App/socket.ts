// import { io, Socket } from 'socket.io-client';

// class SocketService {
//   private socket: Socket | null = null;
//   url = 'http://localhost:8080';

//   connect = (options: Record<string, unknown> = {}): void => {
//     if (!this.socket) {
//       this.socket = io(this.url, options);

//       console.log('ðŸš€ ~ SocketService ~ this.socket.on ~ this.socket?.id2:', this.socket?.id);
//       this.socket.on('connect', () => {
//         console.log('ðŸš€ ~ SocketService ~ this.socket.on ~ id:', this.socket?.id);
//       });

//       this.socket.on('disconnect', () => {
//         console.log('ï¿½ï¿½ï¿½ ~ SocketService ~ this.socket.on ~ disconnect');
//       });

//       this.socket.on('connect_error', (err: Error) => {
//         console.error('Socket connection error:', err);
//       });
//     }
//   };

//   emit = (event: string, data: unknown): void => {
//     if (this.socket) {
//       this.socket.emit(event, data);
//     } else {
//       console.error('Socket is not initialized. Call connect() first.');
//     }
//   };

//   on<T>(event: string, callback: (data: T) => void): void {
//     if (this.socket) {
//       this.socket.on(event, callback);
//     } else {
//       console.error('Socket is not initialized. Call connect() first.');
//     }
//   }

//   off(event: string, callback?: (data: unknown) => void): void {
//     if (this.socket) {
//       this.socket.off(event, callback);
//     } else {
//       console.error('Socket is not initialized. Call connect() first.');
//     }
//   }

//   disconnect(): void {
//     if (this.socket) {
//       this.socket.disconnect();
//       this.socket = null;
//     }
//   }
// }

// const socketService = new SocketService();
// export default socketService;
