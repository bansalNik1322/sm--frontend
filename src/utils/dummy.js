// import React from 'react';
import {
  AiOutlineAreaChart,
  AiOutlineBarChart,
  AiOutlineCalendar,
  AiOutlineShoppingCart,
  AiOutlineStock,
} from 'react-icons/ai';
import { BiColorFill } from 'react-icons/bi';
import { BsBarChart, BsKanban } from 'react-icons/bs';
import { FiEdit, FiPieChart, FiShoppingBag } from 'react-icons/fi';
import { GiLouvrePyramid } from 'react-icons/gi';
import { IoMdContacts } from 'react-icons/io';
import { RiContactsLine, RiStockLine } from 'react-icons/ri';
// import avatar from './avatar.jpg';
// import avatar2 from './avatar2.jpg';
// import avatar3 from './avatar3.png';
// import avatar4 from './avatar4.jpg';
// import product1 from './product1.jpg';
// import product2 from './product2.jpg';
// import product3 from './product3.jpg';
// import product4 from './product4.jpg';
// import product5 from './product5.jpg';
// import product6 from './product6.jpg';
// import product7 from './product7.jpg';
// import product8 from './product8.jpg';

// export const gridOrderImage = (props) => (
//   <div>
//     <img
//       className="rounded-xl h-20 md:ml-3"
//       src={props.ProductImage}
//       alt="order-item"
//     />
//   </div>
// );

// export const gridOrderStatus = (props) => (
//   <button
//     type="button"
//     style={{ background: props.StatusBg }}
//     className="text-white py-1 px-2 capitalize rounded-2xl text-md"
//   >
//     {props.Status}
//   </button>
// );

export const links = [
  {
    title: 'Dashboard',
    links: [
      {
        name: 'ecommerce',
        icon: <FiShoppingBag />,
      },
    ],
  },

  {
    title: 'Pages',
    links: [
      {
        name: 'orders',
        icon: <AiOutlineShoppingCart />,
      },
      {
        name: 'employees',
        icon: <IoMdContacts />,
      },
      {
        name: 'customers',
        icon: <RiContactsLine />,
      },
    ],
  },
  {
    title: 'Apps',
    links: [
      {
        name: 'calendar',
        icon: <AiOutlineCalendar />,
      },
      {
        name: 'kanban',
        icon: <BsKanban />,
      },
      {
        name: 'editor',
        icon: <FiEdit />,
      },
      {
        name: 'color-picker',
        icon: <BiColorFill />,
      },
    ],
  },
  {
    title: 'Charts',
    links: [
      {
        name: 'line',
        icon: <AiOutlineStock />,
      },
      {
        name: 'area',
        icon: <AiOutlineAreaChart />,
      },

      {
        name: 'bar',
        icon: <AiOutlineBarChart />,
      },
      {
        name: 'pie',
        icon: <FiPieChart />,
      },
      {
        name: 'financial',
        icon: <RiStockLine />,
      },
      {
        name: 'color-mapping',
        icon: <BsBarChart />,
      },
      {
        name: 'pyramid',
        icon: <GiLouvrePyramid />,
      },
      {
        name: 'stacked',
        icon: <AiOutlineBarChart />,
      },
    ],
  },
];
