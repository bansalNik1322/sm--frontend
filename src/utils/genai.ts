'use server';

import { google } from '@ai-sdk/google';
import { generateText } from 'ai';
import { subDays, subHours } from 'date-fns';

export interface IContent {
  role: 'user' | 'assistant';
  content: { type: 'text'; text: string }[];
}

type GoogleModelCacheableId = 'models/gemini-1.5-flash-001' | 'models/gemini-1.5-pro-001';
export const generateRecipe = async (prompt: string) => {
  try {
    const model: GoogleModelCacheableId = 'models/gemini-1.5-pro-001';

    const { text } = await generateText({
      model: google(model),
      messages: [
        {
          role: 'user',
          content: [
            {
              type: 'text',
              text: 'My name is Nikhil Bansal',
            },
          ],
        },
        {
          role: 'assistant',
          content: [
            {
              type: 'text',
              text: 'Nice to meet you, Nikhil Bansal! ðŸ‘‹ Is there anything I can help you with today? ðŸ˜Š',
            },
          ],
        },
        {
          role: 'user',
          content: [
            {
              type: 'text',
              text: prompt,
            },
          ],
        },
      ],
    });

    console.log('ðŸš€ ~ apiKey:', text);
    return text;
  } catch (error) {
    console.error('Error generating recipe:', error);
    throw new Error('Failed to generate recipe. Please try again later.');
  }
};
