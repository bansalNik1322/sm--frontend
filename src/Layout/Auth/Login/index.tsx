'use client';
import { Visibility, VisibilityOff } from '@mui/icons-material';
import AppleIcon from '@mui/icons-material/Apple';
import GoogleIcon from '@mui/icons-material/Google';
import {
  Box,
  Button,
  Checkbox,
  Divider,
  FormControl,
  FormControlLabel,
  IconButton,
  InputAdornment,
  TextField,
  Typography,
} from '@mui/material';
import { Formik } from 'formik';
import Cookies from 'js-cookie';
import Link from 'next/link';
import { useRouter } from 'next/navigation';
import React, { useCallback, useEffect, useState } from 'react';
import * as Yup from 'yup';

import { useLoading, useRequest } from '@/components/App';
import { SuspenseLoader } from '@/components/App/Loader';
import { KEYPAIR, REQUEST } from '@/types/interfaces';
import { toastr, validateAuthentication } from '@/utils/helpers';

const initialValues = {
  email: '',
  password: '',
  rememberme: false,
};

const LoginSchema = Yup.object().shape({
  email: Yup.string().required('Required'),
  password: Yup.string().min(6, 'Too Short!').max(100, 'Too Long!').required('Required'),
});

function Index() {
  const router = useRouter();
  const { request, loading } = useRequest();
  const { ButtonLoader } = useLoading();
  const [viewPassword, setViewPassword] = useState(false);

  const validateToken = useCallback(() => {
    if (validateAuthentication()) {
      if (Cookies.get('rememberme') === '1') return router.push('/settings/accounts');
    }
  }, [router]);

  useEffect(() => {
    validateToken();
  }, [validateToken]);

  const handleSubmit = async (values: KEYPAIR) => {
    console.log(values);
    try {
      const { data, status, message } = (await request('LoginUser', { ...values, userid: values.email })) as {
        data: any;
        status: boolean;
        message: string;
      };
      console.log('ðŸš€ ~ handleSubmit ~ status:', status);
      if (status) {
        Cookies.set('accessToken', data?.authTokens?.accessToken);
        Cookies.set('refreshToken', data?.authTokens?.refreshToken);
        return router.push('/dashboard');
      } else {
        toastr('Login Failed', 'error', 'Login');
      }
    } catch (error) {
      console.log('ðŸš€ ~ handleSubmit ~ error:', error);
      throw error;
    }
  };

  if (validateAuthentication()) {
    if (Cookies.get('rememberme') === '1') return <SuspenseLoader />;
  }

  return (
    <Box
      sx={{
        overflow: 'hidden',
        minHeight: '100vh',
        backgroundColor: '#070C27',
        display: 'flex',
        color: 'rgba(203, 204, 210, 0.7)',
        justifyContent: 'center',
        alignItems: 'center',
      }}
    >
      <Box
        sx={{
          backgroundColor: '#111633',
          width: '40%',
          padding: '20px',
          borderRadius: '10px',
          textAlign: 'center',
        }}
      >
        <Typography sx={{ marginBottom: '21px' }} variant="h3" gutterBottom>
          LogIn
        </Typography>

        <Box sx={{ display: 'flex', gap: '20px', justifyContent: 'space-between', marginBottom: '40px' }}>
          <Button fullWidth startIcon={<GoogleIcon />} variant="contained">
            Login With Google
          </Button>
          <Button fullWidth startIcon={<AppleIcon />} variant="contained">
            Login With Apple
          </Button>
        </Box>

        <Divider sx={{ color: '#aeb9cb', margin: '40px 0' }}>
          <span>or Login with</span>
        </Divider>

        <Formik
          initialValues={initialValues}
          validateOnChange={false}
          validateOnBlur={false}
          validationSchema={LoginSchema}
          onSubmit={handleSubmit}
        >
          {({ handleSubmit, handleChange, errors, touched, values }) => (
            <form noValidate onSubmit={handleSubmit}>
              <Box sx={{ marginBottom: '16px' }}>
                <FormControl fullWidth>
                  <TextField
                    label="Email or Username"
                    variant="outlined"
                    name="email"
                    type="email"
                    onChange={handleChange}
                    value={values.email}
                    error={!!errors.email && touched.email}
                    helperText={errors.email && touched.email && errors.email}
                    fullWidth
                  />
                </FormControl>
              </Box>

              <Box sx={{ marginBottom: '16px' }}>
                <FormControl fullWidth>
                  <TextField
                    label="Password"
                    variant="outlined"
                    name="password"
                    type={viewPassword ? 'text' : 'password'}
                    onChange={handleChange}
                    error={!!errors.password && touched.password}
                    helperText={errors.password && touched.password && errors.password}
                    fullWidth
                    InputProps={{
                      endAdornment: (
                        <InputAdornment position="end">
                          <IconButton onClick={() => setViewPassword(!viewPassword)} edge="end">
                            {viewPassword ? <VisibilityOff /> : <Visibility />}
                          </IconButton>
                        </InputAdornment>
                      ),
                    }}
                  />
                </FormControl>
              </Box>

              <Box
                sx={{
                  display: 'flex',
                  justifyContent: 'space-between',
                  marginBottom: '20px',
                  color: 'rgba(203, 204, 210, 0.7)',
                }}
              >
                <FormControlLabel
                  control={<Checkbox name="rememberme" checked={values.rememberme} onChange={handleChange} />}
                  label="Remember me"
                />
                <Link href="/forgot-password" style={{ color: '#bf2fcc' }}>
                  Forgot password
                </Link>
              </Box>

              <Button variant="contained" sx={{ marginBottom: '30px' }} type="submit" fullWidth>
                {loading?.LoginUser_LOADING ? ButtonLoader() : 'Submit'}
              </Button>
            </form>
          )}
        </Formik>

        <Typography variant="body2" color="textSecondary">
          Don't have an Account?{' '}
          <Link href="/register" style={{ color: '#bf2fcc' }}>
            Signup
          </Link>
        </Typography>
      </Box>
    </Box>
  );
}

export default Index;
