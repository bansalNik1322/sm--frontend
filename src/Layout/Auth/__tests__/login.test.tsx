import '@testing-library/jest-dom';

import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { useRouter } from 'next/navigation';

import { useLoading, useRequest } from '../../../components/App';
import { toastr } from '../../../utils/helpers';
import Login from '../Login/index';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

jest.mock('../../../components/App', () => ({
  useRequest: jest.fn(),
  useLoading: jest.fn(),
}));

jest.mock('../../../utils/helpers', () => ({
  toastr: jest.fn(),
  validateAuthentication: jest.fn(),
}));

describe('Login Component Tests', () => {
  const requestMock = jest.fn();
  const pushMock = jest.fn();
  const ButtonLoaderMock = jest.fn(() => <span>Loading...</span>);

  beforeEach(() => {
    (useRequest as jest.Mock).mockReturnValue({
      request: requestMock,
      loading: false,
    });
    (useLoading as jest.Mock).mockReturnValue({
      ButtonLoader: ButtonLoaderMock,
    });
    (useRouter as jest.Mock).mockReturnValue({
      push: pushMock,
    });
    (toastr as jest.Mock).mockReset();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the Login component', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });

  test('should show error message for empty email and password fields', async () => {
    render(<Login />);

    fireEvent.change(screen.getByLabelText(/email or username/i), {
      target: { value: '' },
    });
    fireEvent.change(screen.getByLabelText(/password/i), {
      target: { value: '' },
    });
    const submitButton = screen.getByRole('button', { name: /submit/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const errorMessages = screen.getAllByText('Required');
      console.log('ðŸš€ ~ awaitwaitFor ~ errorMessages:', errorMessages.length);

      expect(errorMessages).toHaveLength(2);
    });
  });

  test('should show error message for invalid email format', async () => {
    render(<Login />);

    fireEvent.change(screen.getByLabelText(/email or username/i), {
      target: { value: 'invalid-email' },
    });

    const submitButton = screen.getByRole('button', { name: /submit/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const emailErrorMessage = screen.getByText('Required');
      console.log('Email Error:', emailErrorMessage.textContent);

      expect(emailErrorMessage).toBeInTheDocument();
    });
  });

  test('should show password length error message', async () => {
    render(<Login />);

    fireEvent.change(screen.getByLabelText(/password/i), {
      target: { value: '123' },
    });

    const submitButton = screen.getByRole('button', { name: /submit/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const passwordErrorMessage = screen.getByText('Too Short!');
      console.log('Password Error:', passwordErrorMessage.textContent);

      expect(passwordErrorMessage).toBeInTheDocument();
    });
  });

  it('calls request API and navigates on successful login', async () => {
    requestMock.mockResolvedValue({
      data: {
        authTokens: {
          accessToken: 'fakeAccessToken',
          refreshToken: 'fakeRefreshToken',
        },
      },
      status: true,
      message: 'Login successful',
    });

    const pushMock = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({ push: pushMock });

    render(<Login />);

    fireEvent.change(screen.getByLabelText(/Email or Username/i), { target: { value: 'silent_knight' } });
    fireEvent.change(screen.getByLabelText(/Password/i), { target: { value: '123456' } });
    const submitButton = screen.getByRole('button', { name: /submit/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(requestMock).toHaveBeenCalledWith('LoginUser', {
        userid: 'silent_knight',
        email: 'silent_knight',
        password: '123456',
        rememberme: false,
      });
      expect(pushMock).toHaveBeenCalledWith('/dashboard');
    });
  });

  it('shows error message on failed login', async () => {
    const failureResponse = {
      status: false,
      message: 'Login Failed',
    };

    const requestMock = jest.fn().mockResolvedValue(failureResponse);

    render(<Login />);

    fireEvent.change(screen.getByLabelText(/Email or Username/i), {
      target: { value: 'test@example.com' },
    });
    fireEvent.change(screen.getByLabelText(/Password/i), {
      target: { value: 'wrongpassword' },
    });

    const submitButton = screen.getByRole('button', { name: /submit/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      //   expect(screen.getByText(/Login Failed/i)).toBeInTheDocument();

      expect(toastr).toHaveBeenCalledWith('Login Failed', 'error', 'Login');
    });

    expect(requestMock).toHaveBeenCalledWith('LoginUser', {
      email: 'test@example.com',
      password: 'wrongpassword',
      userid: 'test@example.com',
      rememberme: false,
    });
  });
});
