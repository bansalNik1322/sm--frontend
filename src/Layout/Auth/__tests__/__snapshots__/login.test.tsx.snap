// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Tests renders the Login component 1`] = `
<div>
  <div
    class="MuiBox-root css-13bwyqb"
  >
    <div
      class="MuiBox-root css-wd8o48"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1rix75w-MuiTypography-root"
      >
        LogIn
      </h3>
      <div
        class="MuiBox-root css-4sehry"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GoogleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
              />
            </svg>
          </span>
          Login With Google
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AppleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.71 19.5c-.83 1.24-1.71 2.45-3.05 2.47-1.34.03-1.77-.79-3.29-.79-1.53 0-2 .77-3.27.82-1.31.05-2.3-1.32-3.14-2.53C4.25 17 2.94 12.45 4.7 9.39c.87-1.52 2.43-2.48 4.12-2.51 1.28-.02 2.5.87 3.29.87.78 0 2.26-1.07 3.81-.91.65.03 2.47.26 3.64 1.98-.09.06-2.17 1.28-2.15 3.81.03 3.02 2.65 4.03 2.68 4.04-.03.07-.42 1.44-1.38 2.83M13 3.5c.73-.83 1.94-1.46 2.94-1.5.13 1.17-.34 2.35-1.04 3.19-.69.85-1.83 1.51-2.95 1.42-.15-1.15.41-2.35 1.05-3.11z"
              />
            </svg>
          </span>
          Login With Apple
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-sdh8yp-MuiDivider-root"
        role="separator"
      >
        <span
          class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
        >
          <span>
            or Login with
          </span>
        </span>
      </div>
      <form
        novalidate=""
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Email or Username
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="email"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Email or Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1uvdljl"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="rememberme"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Remember me
            </span>
          </label>
          <a
            href="/forgot-password"
            style="color: rgb(191, 47, 204);"
          >
            Forgot password
          </a>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1max977-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Don't have an Account?
         
        <a
          href="/register"
          style="color: rgb(191, 47, 204);"
        >
          Signup
        </a>
      </p>
    </div>
  </div>
</div>
`;
